gdx/src/com/badlogic/gdx/math/Intersector.java

```diff
@@ -33,11 +33,49 @@ public final class Intersector {
 	private Intersector () {
 	}
 
-	private final static Vector3 v0 = new Vector3();
-	private final static Vector3 v1 = new Vector3();
-	private final static Vector3 v2 = new Vector3();
-	private final static FloatArray floatArray = new FloatArray();
-	private final static FloatArray floatArray2 = new FloatArray();
+	/** ClassX: thread-safety support
+	 * @author dar */
+	static class IntersectorLocal extends ThreadLocal<IntersectorData> {
+		/*
+		 * @see java.lang.ThreadLocal#initialValue()
+		 */
+		@Override
+		protected IntersectorData initialValue () {
+			return new IntersectorData();
+		}
+	}
+
+	/** ClassX: thread-safety support
+	 * @author dar */
+	static class IntersectorData {
+		final Vector3 v0 = new Vector3();
```

Comment:
Can you not make all these fields `private`?

Reply:
improvement @dar-dev ?

Reply:
here I kept the visibility as before PR but yes, I can make these fields `private`